src/status_im/contexts/profile/events.cljs

```diff
@@ -77,21 +77,30 @@
  :profile/get-profiles-overview-success
  (fn [{:keys [db]}
       [{accounts                               :accounts
-        {:keys [userConfirmed enabled userID]} :centralizedMetricsInfo}]]
-   (let [profiles          (reduce-profiles accounts)
-         profiles-key-uids (keys profiles)
-         new-db            (cond-> db
-                             :always
-                             (assoc :centralized-metrics/user-confirmed? userConfirmed
-                                    :centralized-metrics/enabled?        enabled
-                                    :centralized-metrics/user-id         userID)
+        {:keys [userConfirmed enabled userID]} :centralizedMetricsInfo}
+       {:keys [logout-phase?]}]]
+   (let [profiles                (reduce-profiles accounts)
+         profiles-key-uids       (keys profiles)
+         recently-opened-profile (when-not logout-phase?
+                                   (:key-uid (first (sort-by :timestamp > (vals profiles)))))
+         new-db                  (cond-> db
+                                   :always
+                                   (assoc :centralized-metrics/user-confirmed? userConfirmed
+                                          :centralized-metrics/enabled?        enabled
+                                          :centralized-metrics/user-id         userID)
 
-                             (seq profiles)
-                             (assoc :profile/profiles-overview profiles))]
+                                   (seq profiles)
+                                   (assoc :profile/profiles-overview profiles))]
      {:db new-db
-      :fx [[:dispatch [:profile/get-profiles-auth-method profiles-key-uids]]
+      :fx [(when-not logout-phase?
+             [:dispatch [:profile/profile-selected recently-opened-profile]])
+           (when-not logout-phase?
+             [:dispatch [:profile.login/login-with-biometric-if-available recently-opened-profile]])
```

Comment:
Is it possible to refactor these into a single event that performs both in order to encapsulate the login logic when it's not the `logout-phase`? it'd also remove the need of using two `logout-phase?` checks

Reply:
Thank you, updated.